checkov/terraform/deep_analysis_plan_graph_manager.py

```diff
@@ -5,45 +5,70 @@
 from checkov.terraform.graph_builder.graph_components.block_types import BlockType
 from checkov.common.output.report import Report
 from checkov.terraform.plan_parser import TF_PLAN_RESOURCE_ADDRESS
-from typing import Dict
+from typing import Dict, Tuple
 
 
 class DeepAnalysisGraphManager:
     def __init__(self, tf_graph: TerraformLocalGraph, tf_plan_graph: TerraformLocalGraph) -> None:
         self.tf_graph: TerraformLocalGraph = tf_graph
         self.tf_plan_graph: TerraformLocalGraph = tf_plan_graph
-        self._address_to_tf_vertex_map: Dict[str, TerraformBlock] = {}
-        self._address_to_tf_plan_vertex_map: Dict[str, TerraformBlock] = {}
+        self._address_to_tf_idx_and_vertex_map: Dict[str, Tuple[TerraformBlock, int]] = {}
+        self._address_to_tf_plan_idx_and_vertex_map: Dict[str, Tuple[TerraformBlock, int]] = {}
         self._apply_address_mapping()
 
     def _apply_address_mapping(self) -> None:
-        self._address_to_tf_vertex_map = {
-            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: vertex
-            for vertex in self.tf_graph.vertices
+        self._address_to_tf_idx_and_vertex_map = {
+            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: (i, vertex)
+            for i, vertex in enumerate(self.tf_graph.vertices)
             if vertex.block_type == BlockType.RESOURCE
         }
-        self._address_to_tf_plan_vertex_map = {
-            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: vertex
-            for vertex in self.tf_plan_graph.vertices
+        self._address_to_tf_plan_idx_and_vertex_map = {
+            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: (i, vertex)
+            for i, vertex in enumerate(self.tf_plan_graph.vertices)
             if vertex.block_type == BlockType.RESOURCE
         }
 
+    def append_vertex_to_terraform_graph(self, tf_plan_vertex, tf_plan_vertex_index, address):
+        new_vertex_idx = len(self.tf_graph.vertices)
+        self.tf_graph.vertices.append(tf_plan_vertex)
+        self._address_to_tf_idx_and_vertex_map[address] = (new_vertex_idx, tf_plan_vertex)
+
+        def __get_tf_vertex_idx_from_tf_plan_vertex(v):
+            v = self._address_to_tf_idx_and_vertex_map.get(v.attributes.get('__address__'))
+            if not v:
+                return None
+            return v[1]
```

Comment:
I don't think this needs to be a nested method, just move it to the same level as the others.